src/libLookup/Lookup.cpp

```diff
@@ -1828,6 +1852,117 @@ bool Lookup::ProcessSetOfflineLookups(
     return true;
 }
 
+bool Lookup::ProcessRaiseStartPoW(
+    [[gnu::unused]] const vector<unsigned char>& message,
+    [[gnu::unused]] unsigned int offset, [[gnu::unused]] const Peer& from)
+{
+    // Message = empty
+
+    LOG_MARKER();
+
+#ifdef IS_LOOKUP_NODE
+    // DS leader has informed me that it's time to start PoW
+    m_receivedRaiseStartPoW = true;
+    cv_startPoWSubmission.notify_all();
+
+    LOG_EPOCH(
+        INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+        "Threads running ProcessGetStartPoWFromSeed notified to start PoW");
+
+    // Sleep for a while, then let all remaining threads running ProcessGetStartPoWFromSeed know that it's too late to do PoW
+    // Sleep time = time it takes for new node to try getting DSInfo + actual PoW window
+    this_thread::sleep_for(
+        chrono::seconds(NEW_NODE_SYNC_INTERVAL + POW_WINDOW_IN_SECONDS));
+    m_receivedRaiseStartPoW = false;
+    cv_startPoWSubmission.notify_all();
+
+    LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "Threads running ProcessGetStartPoWFromSeed notified it's too "
+              "late to start PoW");
+
+#endif // IS_LOOKUP_NODE
+
+    return true;
+}
+
+bool Lookup::ProcessGetStartPoWFromSeed(
+    [[gnu::unused]] const vector<unsigned char>& message,
+    [[gnu::unused]] unsigned int offset, [[gnu::unused]] const Peer& from)
+{
+    // Message = [Peer listen port]
+
+    LOG_MARKER();
+
+#ifdef IS_LOOKUP_NODE
+    // Normally I'll get this message from new nodes at the vacuous epoch
+    // Wait a while if I haven't received RAISESTARTPOW from DS leader yet
+    // Wait time = time it takes to finish the vacuous epoch (or at least part of it) + actual PoW window
+    if (!m_receivedRaiseStartPoW)
+    {
+        std::unique_lock<std::mutex> cv_lk(m_MutexCVStartPoWSubmission);
+
+        if (cv_startPoWSubmission.wait_for(
```

Comment:
Instead of using condition variable, I think it's better to use a container to buffer the early received Get messages, as the queries from new node join could be inestimable large, which may run out the resources in thread pool. You can refer the to design of buffer for `ProcessForwardTransaction`

Reply:
As discussed, will be revised in a separate PR. Issue written up: #511.